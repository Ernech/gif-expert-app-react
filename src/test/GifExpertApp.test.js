import '@testing-library/jest-dom'
import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en el componente <GifExpertApp/>.', () => {
    
    
    test('Debe ser igual a su snapshot', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe de mostrar una lista de categorÃ­as', () => {
        
        const categories = ['Music', 'Crash Bandicoot']
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });

});