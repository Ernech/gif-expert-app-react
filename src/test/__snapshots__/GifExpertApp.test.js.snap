// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifExpertApp/>. Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Music"
      key="Music"
    />
    <GifGrid
      category="Crash Bandicoot"
      key="Crash Bandicoot"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente <GifExpertApp/>. Debe ser igual a su snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
